pilot/pkg/proxy/envoy/v2/pushqueue.go

```diff
@@ -0,0 +1,83 @@
+package v2
+
+import (
+	"sync"
+
+	"istio.io/istio/pilot/pkg/model"
+)
+
+type PushInformation struct {
+	// If not empty, it is used to indicate the event is caused by a change in the clusters.
+	// Only EDS for the listed clusters will be sent.
+	edsUpdatedServices map[string]struct{}
+
+	push *model.PushContext
+
+	full bool
+}
+
+type PushQueue struct {
+	mu          sync.RWMutex
+	connections map[*XdsConnection]*PushInformation
+	order       []*XdsConnection
+	signal      chan struct{}
+}
+
+func NewPushQueue() *PushQueue {
+	return &PushQueue{
+		connections: make(map[*XdsConnection]*PushInformation),
+		signal:      make(chan struct{}),
+	}
+}
+
+func (p *PushQueue) Add(proxy *XdsConnection, pushInfo *PushInformation) {
```

Comment:
what if multi pushInfo added for a single proxy?
Let's say the previous one is being processed, what's the worst result?